sdk/python/kfp/dsl/artifact.py

```diff
@@ -98,129 +46,105 @@ class Artifact(object):
      their components.
   2. At runtime, Artifact objects provide helper function/utilities to access
      the underlying RuntimeArtifact pb message, and provide additional layers
-     of validation to ensure type compatibility.
+     of validation to ensure type compatibility for fields specified in the 
+     instance schema.
   """
 
-  # Name of the Artifact type.
-  TYPE_NAME = 'Artifact'
-  # Property schema.
-  # Example usage:
-  #
-  # PROPERTIES = {
-  #   'span': Property(type=PropertyType.INT),
-  #   # Comma separated of splits for an artifact. Empty string means artifact
-  #   # has no split.
-  #   'split_names': Property(type=PropertyType.STRING),
-  # }
-  PROPERTIES = None
+  TYPE_NAME="Artifact"
+
+  # Dict holding the instance's schema fields used for type checking
+  _metadata_fields = Dict[str, MetadataFieldType]
 
   # Initialization flag to support setattr / getattr behavior.
   _initialized = False
 
+  def _parse_instance_schema(self):
+    """ Parses the instance schema of the Artifact and sets up self._metadata_fields map for type-checking.
+    Expects _instance_schema to be set.
```

Comment:
nit: consider document the exceptions under "Raises:"

Reply:
Done.

Reply:
Sorry, I didn't link a reference. The format should be
```
Raises:
  RuntimeError ...
  ValueError ....
```
https://google.github.io/styleguide/pyguide.html#doc-function-raises

Reply:
and ideally, the docstring starts with a single line short description. followed by a blank line and then the details.

Reply:
Fixed.